examples/allocation/tinygo/testdata/greet.go

```diff
@@ -72,13 +78,22 @@ func ptrToString(ptr uint32, size uint32) string {
 	}))
 }
 
-// stringToPtr returns a pointer and size pair for the given string in a way
-// compatible with WebAssembly numeric types.
+// stringToPtr returns a pointer and size pair to the underlying bytes of the given
+// string in a way compatible with WebAssembly numeric types. Since Go strings are
+// immutable, the pointer returned by stringToPtr must not be modified.
 func stringToPtr(s string) (uint32, uint32) {
-	buf := []byte(s)
-	ptr := &buf[0]
+	// for an empty string the return value by TinyGo 0.27 is nil, however the
```

Comment:
As we aren't special casing anything I think it is better we say nothing at all here, or tighten it up to one succinct line. At the moment, this comment dominates the file and has little to do with wasm.

If anyone is interested, they could add details about this into the site doc here or a RATIONALE.md in tinymem, including footnotes here https://github.com/tetratelabs/wazero/pull/1436 or leave it.

Reply:
removed